### DISCUSSION TITLE
Genmon remote acces

### DISCUSSION THREAD

**Discussion by Clearn529** [jgyates/genmon#1134] [General]
Hi, I'm new to this & don't know how it all works. I bought a preloaded unit from Pintsize.me. I installed it to my 22kw generac. I just was able to figure out how to save it to my android phone & iPad.  I can only access at home through my wifi. Is it possible to access away from home base ?  Thanks, Chris

**Comment by jgyates** [comment]
The short answer is no, not without some knowledge of setting up a VPN. Remote access to genmon is not within the scope of this project.

The longer answer is that genmon hosts a web server on the raspberry pi and if you have the technical knowlede you can access it outside of your network via a VPN. Information from past threads on VPN with genmon is available here:

https://github.com/jgyates/genmon/issues?q=label%3A%22remote+access%22+

I have modified the title of this thread to better describe the topic.

**Reply by Clearn529** [reply]
I guess the whole point of having the ability to see what my generator is doing especially when I’m away, to be able to see what’s happening.

**Reply by jgyates** [reply]
@Clearn529 

This is a free software project. If you want that functionality you either have to provide it your self or pay for some other solution.

Genmon will send messages via various methods, so if you have email or one of the other add on communications enabled you can get notified of events the occur with your generator. If you want to see the dashboard you have to expose that on your network and the safe way to do this is via VPN.

If you want to view the genmon dashboard outside of your network, there are only two ways to do this:

1) Make changes to your local network (like a VPN) to the raspberry pi is accessable outside of your network. This requires knowledge of your network would potentially expose your network to the outside internet. These types of changes, since they are network specific, need to be made by the owner of the network (not by me).

2) Add server on the internet that relays the status of each genmon installation. This project does not do that as this would require some server to be available and this would require a cost. With this option, genmon would no longer be free software.

**Comment by billpribyl** [comment]
Have a look at tailscale. Install on your Pi and on your remote device. Free and relatively secure.

**Reply by jgyates** [reply]
https://tailscale.com/

**Comment by Clearn529** [comment]
Yah I understand it's free, thank you for that. All this is too technical for me but with step by step, easy to follow instructions, I can do most things.Thanks 
    On Wednesday, June 12, 2024 at 11:54:54 p.m. EDT, jgyates ***@***.***> wrote:  
 
 


@Clearn529

This is a free software project. If you want that functionality you either have to provide it your self or pay for some other solution.

Genmon will send messages via various methods, so if you have email or one of the other add on communications enabled you can get notified of events the occur with your generator. If you want to see the dashboard you have to expose that on your network and the safe way to do this is via VPN.

If you want to view the genmon dashboard outside of your network, there are only two ways to do this:
   
   -    
Make changes to your local network (like a VPN) to the raspberry pi is accessable outside of your network. This requires knowledge of your network would potentially expose your network to the outside internet. These types of changes, since they are network specific, need to be made by the owner of the network (not by me).

   -    
Add server on the internet that relays the status of each genmon installation. This project does not do that as this would require some server to be available and this would require a cost. With this option, genmon would no longer be free software.

**Reply by Clearn529** [reply]
Another question, I added 2 emails for notifications. I started the generac through genmon, am I supposed to get a notification for that or is it just going to be when the power goes out & it starts ?? Thank you

**Reply by jgyates** [reply]
You should get an email notice every time the generator state changes (cranking, starting,, running, stopped, etc).

**Reply by skipfire** [reply]
@Clearn529 did you setup your mail server information in GenMon's settings?

**Comment by Clearn529** [comment]
That's great, too technical for me. Unless I have step by step easy to follow instructions.Thanks 
    On Wednesday, June 12, 2024 at 08:12:37 p.m. EDT, Bill Pribyl ***@***.***> wrote:  
 
 


Have a look at tailscale. Install on your Pi and on your remote device. Free and relatively secure.

**Comment by Clearn529** [comment]
I thought so. I added 2 email addresses to the notifications. Is there another spot I missed ?thanks 
    On Thursday, June 13, 2024 at 09:52:45 a.m. EDT, Jon R. Helms ***@***.***> wrote:  
 
 


@Clearn529 did you setup your mail server information in GenMon's settings?

**Reply by jgyates** [reply]
You need the set the sending email parameters on the settings page. You set the destination email address on the notification page.

**Reply by jgyates** [reply]
Also, the wiki page for tips and tricks gives details on how to setup email on the settings page.

**Reply by jgyates** [reply]
https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon#setup-gmail-for-use-with-genmon
